src/codec.ts

```diff
@@ -231,6 +248,78 @@ export class PGNumeric {
   }
 }
 
+/**
+ * @typedef ProtoMessage
+ * @see Spanner.protoMessage
+ */
+export class ProtoMessage {
+  value: Buffer;
+  fullName: string;
+  messageFunction?: Function;
+
+  constructor(protoMessageParams: IProtoMessageParams) {
+    this.fullName = protoMessageParams.fullName;
+
+    if (protoMessageParams.value instanceof Buffer) {
+      this.value = protoMessageParams.value;
+    } else if (protoMessageParams.messageFunction) {
+      this.value = protoMessageParams.messageFunction['encode'](
+        protoMessageParams.value
+      ).finish();
+    } else {
+      throw new GoogleError(`protoMessageParams cannot be used for constructing ProtoMessage. 
+          Please pass serializedBytes as value or message object with messageFunction`);
+    }
+
+    this.messageFunction = protoMessageParams.messageFunction;
+  }
+
+  toJSON(): string {
+    if (this.messageFunction != null) {
+      return this.messageFunction['toObject'](
+        this.messageFunction['decode'](this.value)
+      );
+    }
+    return this.value.toString();
+  }
+}
+
+/**
+ * @typedef ProtoEnum
+ * @see Spanner.protoEnum
+ */
+export class ProtoEnum {
+  value: string;
+  fullName: string;
+  enumObject?: object;
+
+  constructor(protoEnumParams: IProtoEnumParams) {
+    this.fullName = protoEnumParams.fullName;
+
+    if (is.number(protoEnumParams.value)) {
```

Comment:
If type is already defined as string for value, why do we need to convert it to string again?

Reply:
I think we need to convert the type for value to number|string